packages/smooth_app/lib/pages/prices/prices_locations_page.dart

```diff
@@ -53,117 +59,94 @@ class _PricesLocationsPageState extends State<PricesLocationsPage>
-                      title: '${item.userCount}',
-                      iconData: PriceButton.userIconData,
-                      tooltip: item.userCount == null
-                          ? null
-                          : appLocalizations.prices_button_count_user(
-                              item.userCount!,
-                            ),
-                    ),
-                    PriceButton(
-                      onPressed: () {},
-                      title: '${item.productCount}',
-                      iconData: PriceButton.productIconData,
-                      tooltip: item.productCount == null
-                          ? null
-                          : appLocalizations.prices_button_count_product(
-                              item.productCount!,
-                            ),
-                    ),
-                    PriceButton(
-                      onPressed: () {},
-                      title: '${item.proofCount}',
-                      iconData: PriceButton.proofIconData,
-                      tooltip: item.proofCount == null
-                          ? null
-                          : appLocalizations.prices_button_count_proof(
-                              item.proofCount!,
-                            ),
-                    ),
-                  ],
-                ),
-              ),
-            );
-          }
-          final String title =
-              appLocalizations.prices_locations_list_length_many_pages(
-            _pageSize,
-            result.total!,
-          );
-          children.insert(
-            0,
-            SmoothCard(child: ListTile(title: Text(title))),
-          );
-          // so that the last content gets not hidden by the FAB
-          children.add(
-            const SizedBox(height: 2 * MINIMUM_TOUCH_SIZE),
-          );
-          return ListView(
-            children: children,
-          );
-        },
+      body: InfiniteScrollList<Location>(
+        manager: _locationManager,
+        itemBuilder: (BuildContext context, Location item) =>
+            _locationManager.buildItem(context: context, item: item, index: 0),
       ),
     );
   }
+}
+
+/// A manager for handling location data with infinite scrolling
+class InfiniteScrollLocationManager extends InfiniteScrollManager<Location> {
+  InfiniteScrollLocationManager({
+    super.initialItems,
+  });
+
+  @override
+  Future<void> fetchData(final int pageNumber) async {
+    final MaybeError<GetLocationsResult> result =
+        await OpenPricesAPIClient.getLocations(
+      GetLocationsParameters()
+        ..pageNumber = pageNumber
+        ..pageSize = 10,
+    );
+    if (result.isError) {
+      throw result.detailError;
+    }
+    final GetLocationsResult value = result.value;
+    updateItems(
+      newItems: value.items ?? <Location>[],
+      pageNumber: value.pageNumber ?? pageNumber,
+      totalItems: value.total,
+      totalPages: value.numberOfPages,
+    );
```

Comment:
```suggestion
    updateItems(
      newItems: value.items,
      pageNumber: value.pageNumber,
      totalItems: value.total,
      totalPages: value.numberOfPages,
    );
```
I would let `updateItems` handle the unexpected cases of null input data.